sqlglot/parser.py

```diff
@@ -4401,6 +4422,10 @@ def _parse_types(
         elif expressions:
             this.set("expressions", expressions)
 
+        # https://materialize.com/docs/sql/types/list/#type-name
+        while self._match(TokenType.LIST):
+            this = exp.DataType(this=exp.DataType.Type.LIST, expressions=[this], nested=True)
```

Comment:
Is this working as intended? I think this tries to parse nested lists but without recursing it seems like it will only match once

Reply:
LGTM - this creates nested list types. E.g. `INT LIST LIST` will be parsed as `LIST(LIST(INT))`.